traveler_ui_kit/src/main/java/com/guestlogix/traveleruikit/activities/BookingSearchActivity.java

```diff
@@ -1,82 +1,54 @@
-import androidx.appcompat.widget.SearchView;
-import androidx.fragment.app.Fragment;
-import androidx.fragment.app.FragmentTransaction;
+import androidx.appcompat.widget.AppCompatEditText;
+import androidx.recyclerview.widget.LinearLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
 
-import com.guestlogix.travelercorekit.callbacks.BookingSearchCallback;
-import com.guestlogix.travelercorekit.models.BookingItemQuery;
-import com.guestlogix.travelercorekit.models.BookingItemSearchResult;
-import com.guestlogix.travelercorekit.models.Traveler;
+import com.guestlogix.travelercorekit.models.BookingItemCategory;
 import com.guestlogix.traveleruikit.R;
-import com.guestlogix.traveleruikit.fragments.BookingSearchResultFragment;
-import com.guestlogix.traveleruikit.fragments.LoadingFragment;
-import com.guestlogix.traveleruikit.fragments.RetryFragment;
-import com.guestlogix.traveleruikit.utils.FragmentTransactionQueue;
 
-public class BookingSearchActivity extends AppCompatActivity
-        implements BookingSearchCallback, RetryFragment.InteractionListener {
+import java.util.ArrayList;
 
-    private FragmentTransactionQueue transactionQueue;
-    private String queryText;
+public class BookingSearchActivity extends AppCompatActivity {
+
+    AppCompatEditText etSearch;
+    AppCompatEditText etLocation;
+    RecyclerView rvCategories;
+    CategoryAdapter categoryAdapter;
 
     @Override
     protected void onCreate(@Nullable Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
 
         setContentView(R.layout.activity_booking_search);
 
-        SearchView searchView = findViewById(R.id.search_bar);
-        searchView.setOnQueryTextListener(new SearchView.OnQueryTextListener() {
-
-            @Override
-            public boolean onQueryTextSubmit(String query) {
-                queryText = query;
-                searchBookings();
-                return false;
-            }
-
-            @Override
-            public boolean onQueryTextChange(String newText) {
-                return false;
-            }
-        });
-
-        this.transactionQueue = new FragmentTransactionQueue(getSupportFragmentManager());
-    }
-
-    private void searchBookings() {
-        Fragment loadingFragment = new LoadingFragment();
-        FragmentTransaction transaction = transactionQueue.newTransaction();
-        transaction.replace(R.id.booking_items_container, loadingFragment);
-        transactionQueue.addTransaction(transaction);
-
-        BookingItemQuery query = new BookingItemQuery(queryText, null, null);
-        Traveler.searchBookingItems(query, this);
+        etSearch = findViewById(R.id.etSearch);
+        etLocation = findViewById(R.id.etLocation);
+        rvCategories = findViewById(R.id.rvCategories);
+
+        findViewById(R.id.btnSearch).setOnClickListener(v -> showResultActivity());
+
+        ArrayList<CategoryAdapter.CategoryItem> lstCategories = new ArrayList<>();
```

Comment:
Can do it this way so you're not maintaining this separately: https://stackoverflow.com/a/16149165

Reply:
yea i know about that and i wanted to change it :D it was actually only a test code :/ forgot to change it. thank you man you review the code so well.